awx/main/models/credential/__init__.py

```diff
@@ -528,9 +528,14 @@ class TowerNamespace:
 
         if 'INVENTORY_UPDATE_ID' not in env:
             # awx-manage inventory_update does not support extra_vars via -e
-            extra_vars = {}
-            for var_name, tmpl in self.injectors.get('extra_vars', {}).items():
-                extra_vars[var_name] = sandbox_env.from_string(tmpl).render(**namespace)
+            def build_extra_vars(node):
+                if isinstance(node, dict):
+                    return {sandbox_env.from_string(k).render(**namespace):
```

Comment:
my 2 cents, I'd rather call `build_extra_vars(k)` as opposed to `sandbox_env.from_string(k).render(**namespace)`. If we're recursive anyway, might as well own it.